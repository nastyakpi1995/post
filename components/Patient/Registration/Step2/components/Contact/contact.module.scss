.validation-contact {
  padding-top: 40px;

  .validation-description {
    font-family: Avenir Next;
    font-style: normal;
    font-weight: 600;
    font-size: 16px;
    line-height: 20px;
    color: #252f4d;
    margin: 0;
  }
  .contact-wrapper {
    display: flex;
    margin-top: 20px;

    .contact-country {
      width: 48%;
      margin-right: 4%;

      .helper {
        font-family: Avenir Next;
        font-style: normal;
        font-weight: 500;
        font-size: 10px;
        letter-spacing: 1px;
        text-transform: uppercase;
        color: #bbbecd;
      }
      .select {
        width: 100%;
        border: 1px solid #ebf1f6;
        height: 44px;
        box-sizing: border-box;
        border-radius: 8px;
        background: #fff;
        padding: 0 14px;
        font-family: Avenir Next;
        font-style: normal;
        font-weight: normal;
        font-size: 16px;
        color: #252f4d;
      }
    }
    .contact-city {
      width: 48%;

      .helper {
        font-family: Avenir Next;
        font-style: normal;
        font-weight: 500;
        font-size: 10px;
        letter-spacing: 1px;
        text-transform: uppercase;
        color: #bbbecd;
      }
      .select {
        width: 100%;
        border: 1px solid #ebf1f6;
        height: 44px;
        box-sizing: border-box;
        border-radius: 8px;
        background: #fff;
        padding: 0 14px;
        font-family: Avenir Next;
        font-style: normal;
        font-weight: normal;
        font-size: 16px;
        color: #252f4d;
      }
    }
  }
  .contact-direction {
    margin-top: 20px;

    .helper {
      font-family: Avenir Next;
      font-style: normal;
      font-weight: 500;
      font-size: 10px;
      letter-spacing: 1px;
      text-transform: uppercase;
      color: #bbbecd;
    }
  }
  .contact-optional {
    display: flex;
    margin-top: 20px;

    .contact-house {
      width: 31%;
      margin-right: 3.5%;

      .helper {
        font-family: Avenir Next;
        font-style: normal;
        font-weight: 500;
        font-size: 10px;
        letter-spacing: 1px;
        text-transform: uppercase;
        color: #bbbecd;
      }
    }
    .contact-apartment {
      width: 31%;
      margin-right: 3.5%;

      .helper {
        font-family: Avenir Next;
        font-style: normal;
        font-weight: 500;
        font-size: 10px;
        letter-spacing: 1px;
        text-transform: uppercase;
        color: #bbbecd;
      }
    }
    .contact-floor {
      width: 31%;

      .helper {
        font-family: Avenir Next;
        font-style: normal;
        font-weight: 500;
        font-size: 10px;
        letter-spacing: 1px;
        text-transform: uppercase;
        color: #bbbecd;
      }
    }
  }
  .contact-zip {
    margin-top: 20px;

    .helper {
      font-family: Avenir Next;
      font-style: normal;
      font-weight: 500;
      font-size: 10px;
      letter-spacing: 1px;
      text-transform: uppercase;
      color: #bbbecd;
    }
  }
  .contact-email {
    margin-top: 20px;

    .helper {
      font-family: Avenir Next;
      font-style: normal;
      font-weight: 500;
      font-size: 10px;
      letter-spacing: 1px;
      text-transform: uppercase;
      color: #bbbecd;
    }
  }
}
