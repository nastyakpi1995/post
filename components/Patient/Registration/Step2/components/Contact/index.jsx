import React from 'react';
import Input from '../../../../../view/ui/Input';
import styles from './style.module.scss';

export default function Contact() {
  return (
    <div className={styles['validation-contact']}>
      <p className={styles['validation-description']}>Contact information</p>
      <div className={styles['contact-wrapper']}>
        <div className={styles['contact-country']}>
          <label className={styles.helper}>
            country
            <select className={styles.select} id="country">
              <option>Belgium</option>
              <option>Ukraine</option>
            </select>
          </label>
        </div>
        <div className={styles['contact-city']}>
          <label className={styles.helper}>
            city
            <select className={styles.select} id="city">
              <option>Brussels</option>
              <option>Kyiv</option>
            </select>
          </label>
        </div>
      </div>
      <div className={styles['contact-direction']}>
        <label className={styles.helper}>
          direction
          <Input
            type="text"
            placeholder="Rue Montagne aux Herbes PotagÃ¨res"
            id="direction"
          />
        </label>
      </div>
      <div className={styles['contact-optional']}>
        <div className={styles['contact-house']}>
          <label className={styles.helper}>
            house number
            <Input type="number" placeholder="26" id="house" />
          </label>
        </div>
        <div className={styles['contact-apartment']}>
          <label className={styles.helper}>
            apartment number
            <Input type="number" placeholder="3" id="apartment" />
          </label>
        </div>
        <div className={styles['contact-floor']}>
          <label className={styles.helper}>
            floor (optional)
            <Input type="number" placeholder="2" id="floor" />
          </label>
        </div>
      </div>
      <div className={styles['contact-zip']}>
        <label className={styles.helper}>
          zip code
          <Input type="number" placeholder="1000" id="zipCode" />
        </label>
      </div>
      <div className={styles['contact-email']}>
        <label className={styles.helper}>
          email
          <Input type="text" placeholder="rowe.shana@mina.biz" id="email" />
        </label>
      </div>
    </div>
  );
}
