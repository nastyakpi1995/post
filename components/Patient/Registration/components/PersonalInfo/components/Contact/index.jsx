import React from 'react';
import types from 'prop-types';
import { withTranslation } from 'i18n';
import { getCountries, getCities } from 'countries-cities';
import {
  Input,
  FieldLabel,
  Select,
  ErrorMessage,
} from '../../../../../../view/ui';
import styles from './style.module.scss';

function Contact({
  personalInfoErrors,
  values,
  errors,
  touched,
  handleChange,
  handleBlur,
  t,
  setFieldValue,
  onResetErrors,
}) {
  return (
    <div className={styles['validation-contact']}>
      <p className={styles['validation-description']}>
        {t('personalInfo.contact.title')}
      </p>
      <div className={styles['contact-wrapper']}>
        <div className={styles['contact-country']}>
          <FieldLabel text={t('personalInfo.contact.country')}>
            <Select
              options={getCountries().map((item) => ({
                value: item,
                label: item,
              }))}
              name="country"
              onChange={({ value, label }) => {
                setFieldValue('country', value);
                setFieldValue('city', getCities(value));
              }}
              defaultValue={{ value: values.country, label: values.country }}
              getOptionValue={() => ({
                value: values.country,
                label: values.country,
              })}
            />
          </FieldLabel>
        </div>
        <div className={styles['contact-city']}>
          <FieldLabel text={t('personalInfo.contact.city')}>
            <Select
              options={getCities(values.country).map((item) => ({
                value: item,
                label: item,
              }))}
              name="city"
              onChange={({ value, label }) => setFieldValue('city', value)}
              defaultValue={{ value: values.city, label: values.city }}
              getOptionValue={() => ({
                value: values.city,
                label: values.city,
              })}
            />
          </FieldLabel>
        </div>
      </div>
      <div className={styles['contact-direction']}>
        <FieldLabel text={t('personalInfo.contact.direction')}>
          <Input
            type="text"
            placeholder="Direction"
            name="direction"
            value={values.direction}
            onChange={handleChange}
            onBlur={handleBlur}
            error={
              errors.direction && touched.direction ? errors.direction : null
            }
          />
          {errors.direction && touched.direction ? (
            <ErrorMessage text={errors.direction} />
          ) : null}
        </FieldLabel>
      </div>
      <div className={styles['contact-optional']}>
        <div className={styles['contact-house']}>
          <FieldLabel text={t('personalInfo.contact.houseNumber')}>
            <Input
              type="text"
              placeholder="1a"
              name="houseNumber"
              value={values.houseNumber}
              onChange={handleChange}
              onBlur={handleBlur}
              error={
                errors.houseNumber && touched.houseNumber
                  ? errors.houseNumber
                  : null
              }
            />
            {errors.houseNumber && touched.houseNumber ? (
              <ErrorMessage text={errors.houseNumber} />
            ) : null}
          </FieldLabel>
        </div>
        <div className={styles['contact-apartment']}>
          <FieldLabel text={t('personalInfo.contact.apartmentNumber')}>
            <Input
              type="text"
              placeholder="2b"
              name="apartmentNumber"
              value={values.apartmentNumber}
              onChange={handleChange}
              onBlur={handleBlur}
              error={
                errors.apartmentNumber && touched.apartmentNumber
                  ? errors.apartmentNumber
                  : null
              }
            />
            {errors.apartmentNumber && touched.apartmentNumber ? (
              <ErrorMessage text={errors.apartmentNumber} />
            ) : null}
          </FieldLabel>
        </div>
        <div className={styles['contact-floor']}>
          <FieldLabel text={t('personalInfo.contact.floorNumber')}>
            <Input
              type="number"
              name="floor"
              placeholder="3"
              value={values.floor}
              onChange={handleChange}
              onBlur={handleBlur}
              error={errors.floor && touched.floor ? errors.floor : null}
            />
            {errors.floor && touched.floor ? (
              <ErrorMessage text={errors.floor} />
            ) : null}
          </FieldLabel>
        </div>
      </div>
      <div className={styles['contact-zip']}>
        <FieldLabel text={t('personalInfo.contact.zipCode')}>
          <Input
            type="text"
            name="zipCode"
            placeholder="1000"
            value={values.zipCode}
            onChange={(e) => {
              if (personalInfoErrors !== undefined) {
                if (personalInfoErrors.zipCode !== undefined) {
                  onResetErrors({
                    zipCode: [],
                  });
                }
              }
              handleChange(e);
            }}
            onBlur={handleBlur}
            error={errors.zipCode && touched.zipCode ? errors.zipCode : null}
          />
          {errors.zipCode && touched.zipCode ? (
            <ErrorMessage text={errors.zipCode} />
          ) : null}
          {personalInfoErrors
            ? personalInfoErrors.zipCode
              ? personalInfoErrors.zipCode.map((el) => {
                  return <ErrorMessage key={el} text={el} />;
                })
              : null
            : null}
        </FieldLabel>
      </div>
      <div className={styles['contact-email']}>
        <FieldLabel text={t('personalInfo.contact.email')}>
          <Input
            type="email"
            name="email"
            value={values.email}
            onChange={handleChange}
            placeholder="name@email.com"
            onBlur={handleBlur}
            error={errors.email && touched.email ? errors.email : null}
          />
          {errors.email && touched.email ? (
            <ErrorMessage text={errors.email} />
          ) : null}
        </FieldLabel>
      </div>
    </div>
  );
}

Contact.propTypes = {
  values: types.shape({
    country: types.any,
    city: types.any,
    direction: types.string,
    houseNumber: types.oneOfType([types.string, types.number]),
    apartmentNumber: types.oneOfType([types.string, types.number]),
    floorNumber: types.oneOfType([types.string, types.number]),
    zipCode: types.oneOfType([types.string, types.number]),
    email: types.string,
  }),
  errors: types.shape({
    country: types.string,
    city: types.string,
    direction: types.string,
    houseNumber: types.string,
    apartmentNumber: types.string,
    floorNumber: types.string,
    zipCode: types.string,
    email: types.string,
  }),
  touched: types.shape({
    country: types.string,
    city: types.string,
    direction: types.string,
    houseNumber: types.string,
    apartmentNumber: types.string,
    floorNumber: types.string,
    zipCode: types.string,
    email: types.string,
  }),
  handleChange: types.func,
  handleBlur: types.func,
  t: types.func,
};

export default withTranslation('registration')(Contact);
