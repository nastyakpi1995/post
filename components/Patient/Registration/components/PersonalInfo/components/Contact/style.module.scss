.validation-contact {
  padding-top: 40px;

  & input::-webkit-outer-spin-button,
  & input::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }

  & input[type=number] {
    -moz-appearance: textfield;
  }

  .validation-description {
    font-family: Avenir Next;
    font-style: normal;
    font-weight: 600;
    font-size: 16px;
    line-height: 20px;
    color: #252f4d;
    margin: 0;
  }
  .contact-wrapper {
    display: flex;
    margin-top: 20px;

    .contact-country {
      width: 48%;
      margin-right: 4%;
    }
    .contact-city {
      width: 48%;
    }
  }
  .contact-direction {
    margin-top: 20px;
  }
  .contact-optional {
    display: flex;
    margin-top: 20px;

    .contact-house {
      width: 31%;
      margin-right: 3.5%;
    }
    .contact-apartment {
      width: 31%;
      margin-right: 3.5%;
    }
    .contact-floor {
      width: 31%;
    }
  }
  .contact-zip {
    margin-top: 20px;
  }
  .contact-email {
    margin-top: 20px;
  }
}
