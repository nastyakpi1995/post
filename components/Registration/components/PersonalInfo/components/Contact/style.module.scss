.validation-contact {
  padding-top: 40px;

  & input::-webkit-outer-spin-button,
  & input::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }

  & input[type=number] {
    -moz-appearance: textfield;
  }

  .validation-description {
    font-family: Avenir Next;
    font-style: normal;
    font-weight: 600;
    font-size: 16px;
    line-height: 20px;
    color: #252f4d;
    margin: 0;
  }
  .contact-wrapper {
    display: flex;
    margin-top: 20px;

    .contact-country {
      width: 48%;
      margin-right: 4%;
    }
    .contact-city {
      width: 48%;
    }
  }
  .contact-direction {
    margin-top: 20px;
  }
  .contact-optional {
    display: flex;
    margin-top: 20px;
    align-items: flex-end;

    .contact-house {
      width: 31%;
      margin-right: 3.5%;
      height: 120px;
      padding-top: 13px;
    }
    .contact-apartment {
      width: 31%;
      margin-right: 3.5%;
      height: 120px;
    }
    .contact-floor {
      width: 31%;
      height: 120px;
    }
  }
  .contact-zip {
    margin-top: 20px;
  }
  .contact-email {
    margin-top: 20px;
  }
  .doc-number {
    display: flex;
    width: 100%;
    margin-top: 20px;

    .doc-number__first {
      width: 100%;

      & input::-webkit-outer-spin-button,
      & input::-webkit-inner-spin-button {
        -webkit-appearance: none;
        margin: 0;
      }

      & input[type=number] {
        -moz-appearance: textfield;
      }
    }
  }
}

@media (max-width: 780px) {
  .contact-house {
    padding-top: 0 !important;
  }
}