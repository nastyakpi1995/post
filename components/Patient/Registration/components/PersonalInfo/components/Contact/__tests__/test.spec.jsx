import React from 'react';
import renderer from 'react-test-renderer';
import Contact from '../index';

it('renders correctly', () => {
  const component = renderer.create(<Contact />).toJSON();
  expect(component).toMatchSnapshot();
});
