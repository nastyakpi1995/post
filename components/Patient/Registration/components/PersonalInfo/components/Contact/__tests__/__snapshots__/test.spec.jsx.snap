// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <p>
    Contact information
  </p>
  <div>
    <div>
      <label>
        country
        <select
          id="country"
        >
          <option>
            Belgium
          </option>
          <option>
            Ukraine
          </option>
        </select>
      </label>
    </div>
    <div>
      <label>
        city
        <select
          id="city"
        >
          <option>
            Brussels
          </option>
          <option>
            Kyiv
          </option>
        </select>
      </label>
    </div>
  </div>
  <div>
    <label>
      direction
      <div
        className="undefined"
      >
        <input
          id="direction"
          name="direction"
          placeholder="Rue Montagne aux Herbes PotagÃ¨res"
          type="text"
        />
      </div>
    </label>
  </div>
  <div>
    <div>
      <label>
        house number
        <div
          className="undefined"
        >
          <input
            id="house"
            name="house"
            placeholder="26"
            type="number"
          />
        </div>
      </label>
    </div>
    <div>
      <label>
        apartment number
        <div
          className="undefined"
        >
          <input
            id="apartment"
            name="apartment"
            placeholder="3"
            type="number"
          />
        </div>
      </label>
    </div>
    <div>
      <label>
        floor (optional)
        <div
          className="undefined"
        >
          <input
            id="floor"
            name="floor"
            placeholder="2"
            type="number"
          />
        </div>
      </label>
    </div>
  </div>
  <div>
    <label>
      zip code
      <div
        className="undefined"
      >
        <input
          id="zipCode"
          name="zipCode"
          placeholder="1000"
          type="number"
        />
      </div>
    </label>
  </div>
  <div>
    <label>
      email
      <div
        className="undefined"
      >
        <input
          id="email"
          name="email"
          placeholder="rowe.shana@mina.biz"
          type="text"
        />
      </div>
    </label>
  </div>
</div>
`;
